src/modules/hip/kernel/resize.hpp

```diff
@@ -0,0 +1,735 @@
+    int srcLocation = (int)ceilf(srcLocationFloat);
+    *weight = srcLocation - srcLocationFloat;
+    *srcLoc = ((srcLocation > limit) ? limit : srcLocation) * srcStride;
+}
+
+// -------------------- Set 1 - Bilinear Interpolation --------------------
+
+template <typename T>
+__global__ void resize_bilinear_pkd_tensor(T *srcPtr,
+                                           uint2 srcStridesNH,
+                                           T *dstPtr,
+                                           uint2 dstStridesNH,
+                                           RpptImagePatchPtr dstImgSize,
+                                           RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint2 dstDimsWH;
+    dstDimsWH.x = dstImgSize[id_z].width;
+    dstDimsWH.y = dstImgSize[id_z].height;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNH.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    int4 srcRoi_i4 = *(int4 *)&roiTensorPtrSrc[id_z];
+    d_float16 locSrc_f16;
+    resize_roi_and_srclocs_hip_compute(&srcRoi_i4, &dstDimsWH, id_x, id_y, &locSrc_f16);
+
+    d_float24 dst_f24;
+    rpp_hip_interpolate24_bilinear_pkd3(srcPtr + srcIdx, srcStridesNH.y, &locSrc_f16, &srcRoi_i4, &dst_f24, false);
+    rpp_hip_pack_float24_pkd3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+__global__ void resize_bilinear_pln_tensor(T *srcPtr,
+                                           uint3 srcStridesNCH,
+                                           T *dstPtr,
+                                           uint3 dstStridesNCH,
+                                           int channelsDst,
+                                           RpptImagePatchPtr dstImgSize,
+                                           RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint2 dstDimsWH;
+    dstDimsWH.x = dstImgSize[id_z].width;
+    dstDimsWH.y = dstImgSize[id_z].height;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNCH.x);
+    uint dstIdx = (id_z * dstStridesNCH.x) + (id_y * dstStridesNCH.z) + id_x;
+
+    int4 srcRoi_i4 = *(int4 *)&roiTensorPtrSrc[id_z];
+    d_float16 locSrc_f16;
+    resize_roi_and_srclocs_hip_compute(&srcRoi_i4, &dstDimsWH, id_x, id_y, &locSrc_f16);
+
+    d_float8 dst_f8;
+    rpp_hip_interpolate8_bilinear_pln1(srcPtr + srcIdx, srcStridesNCH.z, &locSrc_f16, &srcRoi_i4, &dst_f8, false);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+
+    if (channelsDst == 3)
```

Comment:
is there a case for other values of channelsDst?

Reply:
Currently we are only handling cases for channelsDst =1 (PLN1) and channelsDst = 3 (PLN3)